### PR TITLE
Thermocline analysis

### PR DESCRIPTION
**Pull request by jkrasting**

### CODE REVIEW

File: om4labs/diags/stratification/stratification.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+            [dictArgs["argo_temp_file"], dictArgs["argo_psal_file"]],
+            combine="by_coords",
+            decode_times=False,
+        )
+    else:
+        # use dataset from catalog, either from command line or default
+        cat = open_intake_catalog(dictArgs["platform"], "obs")
+        argo_dset = cat["Argo_Climatology"].to_dask()
+
+    # standardize longitude to go from 0 to 360.
```
[the remark this answers is not in the document]
Comment (jkrasting) [reply om4labs/diags/stratification/stratification.py:162]: We can create a common function that will standardize the lon range by searching for the minimum value and the roll the index to make the minimum value the start of the array.

[the remark this answers is not in the document]
Comment (jkrasting) [reply om4labs/diags/stratification/stratification.py:162]: #90 should address this

[the remark this answers is not in the document]
Comment (jkrasting) [reply om4labs/diags/stratification/stratification.py:163]: And see a3efb5a

**Comment by jkrasting** [comment]
@raphaeldussin - this PR depends on #90.  Any other comments?

**Comment by raphaeldussin** [comment]
#90 is merged, can you rebase?

**Comment by jkrasting** [comment]
@raphaeldussin - rebased against your master.

**Comment by raphaeldussin** [comment]
@jkrasting I have merged a PR with a notebook to create the ARGO T/S clim data. When you have a chance, can you test the file works and we should put the file into the common archive so we need to change paths in the intake catalogs

**Comment by jkrasting** [comment]
@raphaeldussin - this works with the data notebook you supplied.

File: om4labs/catalogs/obs_catalog_gfdl.yml [hunk 2]
```diff
@@ -3,6 +3,40 @@ plugins:
     - module: intake_xarray
 sources:
 
+  Argo_Climatology:
+    description: "Gridded in-situ temperature and salinity derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: ['/net2/bgr/Data/ARGO/RG_Clim/Updated/RG_ArgoClim_Temp_Extended_200401-202010.nc', '/net2/bgr/Data/ARGO/RG_Clim/Updated/RG_ArgoClim_Psal_Extended_200401-202010.nc']
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_gfdl.yml:10]: point to merged T/S file here: `/archive/Raphael.Dussin/datasets/Roemmich_Gilson_ARGO_TS_Clim`

File: om4labs/catalogs/obs_catalog_gfdl.yml [hunk 3]
```diff
@@ -3,6 +3,40 @@ plugins:
...
+        decode_times: False
+        combine: "by_coords"
+    metadata:
+      origin_url: ''
+
+  Argo_Temp:
+    description: "Gridded in-situ temperature derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: '/net2/bgr/Data/ARGO/RG_Clim/Updated/RG_ArgoClim_Temp_Extended_200401-202010.nc'
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_gfdl.yml:22]: same

File: om4labs/catalogs/obs_catalog_gfdl.yml [hunk 4]
```diff
@@ -3,6 +3,40 @@ plugins:
...
+      xarray_kwargs:
+        decode_times: False
+    metadata:
+      origin_url: ''
+
+  Argo_Psal:
+    description: "Gridded in-situ practical salinity derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: '/net2/bgr/Data/ARGO/RG_Clim/Updated/RG_ArgoClim_Psal_Extended_200401-202010.nc'
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_gfdl.yml:33]: same

File: om4labs/catalogs/obs_catalog_testing.yml [hunk 5]
```diff
@@ -3,6 +3,40 @@ plugins:
     - module: intake_xarray
 sources:
 
+  Argo_Climatology:
+    description: "Gridded in-situ temperature and salinity derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: ['test_data/obs/RG_ArgoClim_Temp_Extended_200401-202010.nc', 'test_data/obs/RG_ArgoClim_Psal_Extended_200401-202010.nc']
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_testing.yml:10]: probably needs a new copy

File: om4labs/catalogs/obs_catalog_testing.yml [hunk 6]
```diff
@@ -3,6 +3,40 @@ plugins:
...
+        decode_times: False
+        combine: "by_coords"
+    metadata:
+      origin_url: ''
+
+  Argo_Temp:
+    description: "Gridded in-situ temperature derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: 'test_data/obs/RG_ArgoClim_Temp_Extended_200401-202010.nc'
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_testing.yml:22]: merged T and S

File: om4labs/catalogs/obs_catalog_testing.yml [hunk 7]
```diff
@@ -3,6 +3,40 @@ plugins:
...
+      xarray_kwargs:
+        decode_times: False
+    metadata:
+      origin_url: ''
+
+  Argo_Psal:
+    description: "Gridded in-situ practical salinity derived from Argo profiles"
+    driver: netcdf
+    args:
+      urlpath: 'test_data/obs/RG_ArgoClim_Psal_Extended_200401-202010.nc'
```
Comment (raphaeldussin) [inline om4labs/catalogs/obs_catalog_testing.yml:33]: same

File: om4labs/diags/stratification/stratification.py [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+    ----------
+    dictArgs : dict
+        Dictionary containing argparse options
+
+    Returns
+    -------
+    xarray.DataSet
+    """
+
+    tempvar = "thetao"
```
Comment (raphaeldussin) [inline om4labs/diags/stratification/stratification.py:112]: move into kwargs for read?

**Review by raphaeldussin — CHANGES_REQUESTED** [review]

**Comment by jkrasting** [comment]
@raphaeldussin - can you break out the mean fields (`ARGO_TEMPERATURE_MEAN` and `ARGO_SALINITY_MEAN`) into their own file?  The complete file at 5 GB is a bit too much for a speedy analysis on Dora and for the testing tarball.

**Comment by raphaeldussin** [comment]
fair! I have built annual and monthly clim for potential temperature and salinity in the same directory

**Pull request merged**